### DISCUSSION TITLE
Protocols that use UDP

### DISCUSSION THREAD

**Discussion by arinc9** [v2fly/v2ray-core#1745] [General]
Hey everyone.

I want to connect to WireGuard where the firewall blocks packets which include the WireGuard header.
I plan to put my v2ray server in between the client and the wireguard server. I want to change the characteristics of the WireGuard packets a little bit so the firewall won’t detect the WireGuard header on the packets and let them pass through.
I don’t want to encapsulate the WireGuard packets under TCP.

There is SOCKS which can proxy the UDP packets but it needs to send stuff over TCP for the UDP ASSOCIATE command. I don’t want to use TCP at all. StreamSettings as QUIC will solve this though. SOCKS UDP packets are also not encapsulated under QUIC.

Vless looks like a good substitute for SOCKS but it uses TCP. Using QUIC as the transport under streamSettings of vless would encapsulate the TCP packet (which already encapsulates the WireGuard packets) under UDP so this is not preferred.

Is there any protocol on v2ray that natively uses UDP for communication other than SOCKS?
Could vless optionally use UDP in the future? Like: `network: "tcp" | "udp"` for OutboundConfigurationObject and InboundConfigurationObject?

**Comment by xiaokangwang** [comment]
VLite, Shadowsockets have UDP over UDP encrypted proxy support.

**Reply by arinc9** [reply]
VLite looks like a great option. I’ll try it out. I plan to help out translating the remaining English translation of the whole project.

**Reply by Arkad19** [reply]
Is it possible to use it now together with v2ray?

**Reply by arinc9** [reply]
> Is it possible to use it now together with v2ray?

I couldn't make VLite work. I opened an issue but yet to get a response from anyone.

**Reply by Arkad19** [reply]
Yes, it is a pity. There is little information on the Internet. Would be a nice addition to v2ray
